op-e2e/actions/interop/interop_fork_test.go

```diff
@@ -0,0 +1,331 @@
+
+		// Sanity check system state
+		dsl.RequireUnsafeTimeOffset(t, c, 2) // Interop still not enabled
+	}
+
+	// Settle both chains to DA and assert nothing changed
+	superchainSyncAsserter.RequireAllSeqSyncStatuses(func() {
+		system.Actors.ActBatchAndMine(t, dsl.WithMarkFinal())
+		dsl.RequireL1Heads(t, system, 2, 1)
+	})
+
+	// Still pre interop
+	for _, c := range chains {
+		syncAsserter := superchainSyncAsserter.ChainAsserters[c.ChainID]
+
+		// Verify promotion of crossUnsafe, localSafe and safe
+		syncAsserter.RequireSeqSyncStatus(func() {
+			system.ActSyncSupernode(t, dsl.WithChains(c), dsl.WithLatestSignal())
+		}, dsl.WithLocalSafeAdvancesTo(1), dsl.WithSafeAdvancesTo(1)) // Assert unsafe head does not move, but update crossUnsafe, locaSafe and safe heads
+
+		syncAsserter.RequireSeqSyncStatus(func() {
+			c.Sequencer.ActL1FinalizedSignal(t)
+			c.Sequencer.ActL2PipelineFull(t)
+		}, dsl.WithFinalizedAdvancesTo(1)) // assert unsafe head does not move, but update finalized head
+
+		// Interop
+		syncAsserter.RequireSeqSyncStatus(func() {
+			// Build another L2 block so that Interop activates
+			system.AddL2Block(c, dsl.WithL2BlocksUntilTimestamp(*c.Sequencer.RollupCfg.InteropTime))
+		}, dsl.WithUnsafeAdvancesTo(2), dsl.WithCrossUnsafeAdvancesTo(2)) // assert unsafe and crossUnsafe advance by one
+
+		////////////////////////////
+		// After Interop Upgrade (for the current chain in iteration)
+		////////////////////////////
+
+		dsl.RequireUnsafeTimeOffset(t, c, 4) // interop should be enabled
+		// Assert for the first time on supervisor heads
+		dsl.RequireSupervisorChainHeads(t, svr, c, syncAsserter.PrevStatus.UnsafeL2.ID(), syncAsserter.PrevStatus.CrossUnsafeL2.ID(), syncAsserter.PrevStatus.LocalSafeL2.ID(), syncAsserter.PrevStatus.SafeL2.ID(), eth.BlockID{Hash: common.BytesToHash([]byte{}), Number: 0})
+
+		// TODO(15863): The initial state of the finalized head differs between supervisor
+		// and sequencer. In an action test setup, the sequencer considers
+		// genesis head as finalized from the start. The supervisor however sets
+		// the finalized head to a nil value until it it set. Seems worth fixing.
+		// The following will fail.
+		// system.ActSyncSupernode(t, dsl.WithChains(c), dsl.WithFinalizedSignal(), dsl.WithLatestSignal(), dsl.WithRequireFinalizedAdvances())
+		// syncAsserter.RequireSupChainHeadsBySyncStatus()
+	}
+
+	// Settle both chains to DA again and have supervisor ingest
+	system.Actors.ActBatchAndMine(t)
+
+	// Verify that L1 heads actually updated
+	dsl.RequireL1Heads(t, system, 3, 1)
+
+	// // TODO: Why doesn't the cross head promotion happen here?
+	superchainSyncAsserter.RequireAllSeqSyncStatuses(func() {
+		system.ActSyncSupernode(t, dsl.WithLatestSignal())
+	}, dsl.WithMapChainAssertions(dsl.WithLocalSafeAdvancesTo(2)))
+
+	superchainSyncAsserter.RequireAllSeqSyncStatuses(func() {
+		system.ActSyncSupernode(t, dsl.WithLatestSignal())
```

Comment:
From my experiment, `dsl.WithLatestSignal()` is elective here to pass the test. Inside `ActSyncSupernode`, we have
```
	for _, chain := range cfg.Chains {
		// Move supervisor safe head forward by one block
		chain.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
	}
	d.Actors.Supervisor.ProcessFull(t)
	for _, chain := range cfg.Chains {
		chain.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
	}
```
So my theory is that the supervisor may emit the event to itself, and we need additional API calls such as ProcessFull/ActL2PipelineFull to propagate cross-safe heads to the sequencer.

Reply:
Thanks, I removed `dsl.WithLatestSignal()` and the test still seems to pass.